bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/Bookie.java

```diff
@@ -1258,7 +1259,7 @@ private Journal getJournal(long ledgerId) {
     /**
      * Add an entry to a ledger as specified by handle.
      */
-    private void addEntryInternal(LedgerDescriptor handle, ByteBuf entry, WriteCallback cb, Object ctx)
+    private void addEntryInternal(LedgerDescriptor handle, ByteBuf entry, LedgerType ledgerType, WriteCallback cb, Object ctx)
```

Comment:
- addEntryInternal takes a boolean `requiresForce` as parameter
- addEntry => calls addEntryInternal(ledgerType.equals(LedgerType.FORCE_ON_JOURNAL)
- recoveryAddEntry => callss addEmtryInternal(true)

all the recovery adds should force writes, no matter what ledger type it is.

Reply:
done